File: Going Dutch/MCPaymentPresence.m
//
//  MCPaymentPresence.m
//  We all pay
//
//  Created by Mark Cornelisse on 23/07/14.
//  Copyright (c) 2014 Mark Cornelisse. All rights reserved.
//

#import "MCPaymentPresence.h"
#import "MCPayment.h"
#import "MCPerson.h"


@implementation MCPaymentPresence

#pragma mark - NSManagedObject

- (void)awakeFromInsert {
    [super awakeFromInsert];
    [self setPrimitiveValue:NSUUID.UUID.UUIDString forKey:@"uniqueId"];
    NSDate *now = NSDate.date;
    [self setPrimitiveValue:now forKey:@"dateCreated"];
    [self setPrimitiveValue:now forKey:@"dateModified"];
}

#pragma mark - NSObject

@end

File: Going Dutch/MCPaymentPresence.swift
//
//  MCPaymentPresence.swift
//  We all pay
//
//  Created by Mark Cornelisse on 26/08/2025.
//  Copyright © 2025 Mark Cornelisse. All rights reserved.
//

import UIKit

@objc(MCPaymentPresence) public final class MCPaymentPresence: NSManagedObject {
    
    // MARK: NSManagedObject
    
    public override func awakeFromInsert() {
        super.awakeFromInsert()
        
        self.setPrimitiveValue(UUID().uuidString, forKey: #keyPath(MCPaymentPresence.uniqueId))
        let now = Date()
        self.setPrimitiveValue(now, forKey: #keyPath(MCPaymentPresence.dateCreated))
        self.setPrimitiveValue(now, forKey: #keyPath(MCPaymentPresence.dateModified))
    }
    
    // MARK: NSObject
}
